src/Controller/ControllerAbstract.php

```diff
@@ -36,162 +22,127 @@ abstract class ControllerAbstract implements IController
     /** @var IResponse */
     protected $response;
```

Comment:
I'm thinking this and the request should be private and have a protected getter instead. The reason is, this is a constant source of pain when using traits.

Reply:
Or it can stay protected, but have a getter...